server/host_controls.go

```diff
@@ -56,3 +56,40 @@ func (p *Plugin) changeHost(requesterID, channelID, newHostID string) error {
 
 	return nil
 }
+
+func (p *Plugin) muteSession(requesterID, channelID, sessionID string) error {
+	state, err := p.getCallState(channelID, false)
+	if err != nil {
+		return err
+	}
+
+	if state == nil {
+		return errors.New("no call ongoing")
+	}
+
+	if requesterID != state.Call.GetHostID() {
+		if isAdmin := p.API.HasPermissionTo(requesterID, model.PermissionManageSystem); !isAdmin {
+			return errors.New("no permissions to mute session")
+		}
+	}
+
+	ust, ok := state.sessions[sessionID]
+	if !ok {
+		return errors.New("session is not in the call")
+	}
+
+	if p.isBot(ust.UserID) {
+		return errors.New("cannot mute the bot")
+	}
+
+	if !ust.Unmuted {
+		return errors.New("user is already muted")
+	}
```

Comment:
Not sure, maybe I'd keep it idempotent and just return nil. We are not doing this under lock so it's possible this would happen under normal circumstances.

Reply:
Ah good point, done.